packages/dds/tree/src/feature-libraries/sequence-field/sequenceFieldCodecV1.ts

```diff
@@ -285,13 +307,29 @@ export function makeV1Codec(
 				if (mark.cellId !== undefined) {
 					decodedMark.cellId = cellIdCodec.decode(mark.cellId, context.baseContext);
 				}
+				// This assignment is solely to recover the type of `decodedMark`.
```

Comment:
I don't understand what this means.

Reply:
The type system thinks that the type of `decodedMark` is `NoopMark & HasMarkFields`. This prevents the if block below from compiling.
To circumvent that, I am casting `decodedMark` back to a `Mark`.
I was concerned that such a assignment would look weird to anyone reading the code. They might either think that it is unecessary (as it should be) or that it is needed for some very fishy reason that is hard to see.
Let me know if you have suggestions on how to get the message acrross more clearly.